extensions/functions/src/main/java/io/crate/window/OffsetValueFunctions.java

```diff
@@ -89,30 +90,119 @@ public Signature boundSignature() {
     public Object execute(int idxInPartition,
                           WindowFrameState currentFrame,
                           List<? extends CollectExpression<Row, ?>> expressions,
-                          Input... args) {
-        final int offset;
-        if (args.length > 1) {
-            Object offsetValue = args[1].value();
-            if (offsetValue == null) {
-                return null;
+                          boolean ignoreNulls,
+                          Input[] args) {
+        calculateOffsetIfNull(args);
+        if (offset == null) {
+            return null;
+        }
+        try {
+            if (ignoreNulls == false) {
+                return getValueFromTargetIndex(offsetDirection.getTargetIndex(idxInPartition, offset),
+                                               currentFrame,
+                                               expressions,
+                                               args);
+            }
+            if (offsetDirection == OffsetDirection.FORWARD) {
+                return executeLeadIgnoreNulls(idxInPartition, currentFrame, expressions, args);
```

Comment:
I think these methods could become part of the `offsetDirection`.  `offsetDirection.getValueAtOffsetIgnorningNulls(...)`

Reply:
done, because enum being static context, had to convert it to class as well as moving more methods into `offsetDirection` for consistency